src/mediapipe_calculators/pytensor_ovtensor_converter.cc

```diff
@@ -0,0 +1,116 @@
+            cc->Inputs().Tag("OVMS_PY_TENSOR").Set<PyObjectWrapper<py::object>>();
+            cc->Outputs().Tag("OVTENSOR").Set<ov::Tensor>();
+        }
+
+        LOG(INFO) << "PyobjectOvtensorConverterCalculator [Node: " << cc->GetNodeName() << "] GetContract end";
+        return absl::OkStatus();
+    }
+
+    absl::Status Close(CalculatorContext* cc) final {
+        LOG(INFO) << "PyobjectOvtensorConverterCalculator [Node: " << cc->NodeName() << "] Close";
+        return absl::OkStatus();
+    }
+
+    absl::Status Open(CalculatorContext* cc) final {
+        LOG(INFO) << "PyobjectOvtensorConverterCalculator [Node: " << cc->NodeName() << "] Open start";
+        outputTimestamp = mediapipe::Timestamp(mediapipe::Timestamp::Unset());
+        LOG(INFO) << "PyobjectOvtensorConverterCalculator [Node: " << cc->NodeName() << "] Open end";
+        return absl::OkStatus();
+    }
+
+    absl::Status Process(CalculatorContext* cc) final {
+        LOG(INFO) << "PyobjectOvtensorConverterCalculator [Node: " << cc->NodeName() << "] Process start";
+        py::gil_scoped_acquire acquire;
+        if (*(cc->Inputs().GetTags().begin()) == "OVTENSOR") {
+            auto& input_tensor = cc->Inputs().Tag("OVTENSOR").Get<ov::Tensor>();
+            py::object outTensor;
+            std::unique_ptr<PyObjectWrapper<py::object>> outputPtr = std::make_unique<PyObjectWrapper<py::object>>(outTensor);
+            PythonBackend pythonBackend;
+            std::vector<int64> shape;
+            for (const auto& dim : input_tensor.get_shape()) {
+                shape.push_back(dim);
+            }
+            const auto& options = cc->Options<PyobjectOvtensorConverterCalculatorOptions>();
+            const auto tagOutputNameMap = options.tag_to_output_tensor_names();
+            auto outputName = tagOutputNameMap.at("OVMS_PY_TENSOR").c_str();
+            pythonBackend.createOvmsPyTensor(
+                outputName,
+                const_cast<void*>((const void*)input_tensor.data()),
+                shape,
+                toString(ovElementTypeToOvmsPrecision(input_tensor.get_element_type())),
+                input_tensor.get_byte_size(),
+                outputPtr);
+
+            cc->Outputs().Tag("OVMS_PY_TENSOR").Add(outputPtr.release(), cc->InputTimestamp());
+            LOG(INFO) << "PyobjectOvtensorConverterCalculator CONVERSION END1";
+        } else {
+            auto& input_tensor = cc->Inputs().Tag("OVMS_PY_TENSOR").Get<PyObjectWrapper<py::object>>();
+            auto precision = ovmsPrecisionToIE2Precision(fromString(input_tensor.getProperty<std::string>("datatype")));
+            ov::Shape shape;
+            for (const auto& dim : input_tensor.getProperty<std::vector<py::ssize_t>>("shape")) {
+                shape.push_back(dim);
+            }
+            auto data = reinterpret_cast<const void*>(input_tensor.getProperty<void*>("ptr"));
+            ov::Tensor* output = new ov::Tensor(precision, shape, const_cast<void*>(data));
```

Comment:
Does the ov tensor copy the data? If not then once py tensor gets destroyed the buffer will get invalidated.

Reply:
In this case we could leverage similar approach as here with holders:
https://github.com/openvinotoolkit/model_server/blob/main/src/mediapipe_internal/mediapipegraphexecutor.cpp